akka-typed/src/main/scala/akka/typed/javadsl/Actor.scala

```diff
@@ -0,0 +1,243 @@
+   */
+  def same[T]: Behavior[T] = Behavior.same
+
+  /**
+   * Return this behavior from message processing in order to advise the
+   * system to reuse the previous behavior, including the hint that the
+   * message has not been handled. This hint may be used by composite
+   * behaviors that delegate (partial) handling to other behaviors.
+   */
+  def unhandled[T]: Behavior[T] = Behavior.unhandled
+
+  /**
+   * Return this behavior from message processing to signal that this actor
+   * shall terminate voluntarily. If this actor has created child actors then
+   * these will be stopped as part of the shutdown procedure. The PostStop
+   * signal that results from stopping this actor will NOT be passed to the
+   * current behavior, it will be effectively ignored.
+   */
+  def stopped[T]: Behavior[T] = Behavior.stopped
+
+  /**
+   * A behavior that treats every incoming message as unhandled.
+   */
+  def empty[T]: Behavior[T] = Behavior.empty
+
+  /**
+   * A behavior that ignores every incoming message and returns “same”.
+   */
+  def ignore[T]: Behavior[T] = Behavior.ignore
+
+  /**
+   * Construct an actor behavior that can react to incoming messages but not to
+   * lifecycle signals. After spawning this actor from another actor (or as the
+   * guardian of an [[akka.typed.ActorSystem]]) it will be executed within an
+   * [[ActorContext]] that allows access to the system, spawning and watching
+   * other actors, etc.
+   *
+   * This constructor is called immutable because the behavior instance doesn't
+   * have or close over any mutable state. Processing the next message
+   * results in a new behavior that can potentially be different from this one.
+   * State is updated by returning a new behavior that holds the new immutable
+   * state.
+   */
+  def immutable[T](onMessage: JapiFunction2[ActorContext[T], T, Behavior[T]]): Behavior[T] =
+    new BehaviorImpl.ImmutableBehavior((ctx, msg) ⇒ onMessage.apply(ctx, msg))
+
+  /**
+   * Construct an actor behavior that can react to both incoming messages and
+   * lifecycle signals. After spawning this actor from another actor (or as the
+   * guardian of an [[akka.typed.ActorSystem]]) it will be executed within an
+   * [[ActorContext]] that allows access to the system, spawning and watching
+   * other actors, etc.
+   *
+   * This constructor is called immutable because the behavior instance doesn't
+   * have or close over any mutable state. Processing the next message
+   * results in a new behavior that can potentially be different from this one.
+   * State is updated by returning a new behavior that holds the new immutable
+   * state.
+   */
+  def immutable[T](
+    onMessage: JapiFunction2[ActorContext[T], T, Behavior[T]],
+    onSignal:  JapiFunction2[ActorContext[T], Signal, Behavior[T]]): Behavior[T] = {
+    new BehaviorImpl.ImmutableBehavior(
+      (ctx, msg) ⇒ onMessage.apply(ctx, msg),
+      { case (ctx, sig) ⇒ onSignal.apply(ctx, sig) })
+  }
```

Comment:
I tried the same "builder" pattern as in the scaladsl, but that would mean that the following type annotation must be added (and the static import would not work):
```java
return Actor.<String> immutable(
  (ctx, msg) -> logic.onMessage(ctx, msg))
  .onSignal((ctx, sig) -> logic.onSignal(ctx, sig));
```

that's too ugly